### ISSUE TITLE
"Purge Spam" queue jobs failing over and over

### ISSUE THREAD

**Issue by benjaminkohl** [solspace/craft-freeform#360]
With Freeform 4.0.0-beta.8, we are seeing failed queue jobs for "Purge Spam" stacking up in the Queue Manager. Here is a copy of the stack trace. The QueryAbortedException appears to be thrown by Craft without an accompanying message.

```
2022-06-02 14:23:42 [queue.ERROR] [craft\queue\QueueLogBehavior::afterError]  [84041] Purge Spam (attempt: 1, pid: 7842) - Error (time: 0.000s):  {"memory":10764168} 
2022-06-02 14:23:42 [queue.ERROR] [craft\db\QueryAbortedException]  {"trace":["#0 /site/path/site-prod/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\\elements\\db\\ElementQuery->prepare()","#1 /site/path/site-prod/vendor/yiisoft/yii2/db/Query.php(157): yii\\db\\QueryBuilder->build()","#2 /site/path/site-prod/vendor/yiisoft/yii2/db/BatchQueryResult.php(159): yii\\db\\Query->createCommand()","#3 /site/path/site-prod/vendor/yiisoft/yii2/db/BatchQueryResult.php(132): yii\\db\\BatchQueryResult->fetchData()","#4 /site/path/site-prod/vendor/yiisoft/yii2/db/BatchQueryResult.php(121): yii\\db\\BatchQueryResult->next()","#5 /site/path/site-prod/vendor/solspace/craft-freeform/packages/plugin/src/Services/SubmissionsService.php(453): yii\\db\\BatchQueryResult->rewind()","#6 /site/path/site-prod/vendor/solspace/craft-freeform/packages/plugin/src/Jobs/PurgeSpamJob.php(14): Solspace\\Freeform\\Services\\SubmissionsService->purgeSubmissions()","#7 /site/path/site-prod/vendor/yiisoft/yii2-queue/src/Queue.php(246): Solspace\\Freeform\\Jobs\\PurgeSpamJob->execute()","#8 /site/path/site-prod/vendor/yiisoft/yii2-queue/src/cli/Queue.php(147): yii\\queue\\Queue->handleMessage()","#9 /site/path/site-prod/vendor/craftcms/cms/src/queue/Queue.php(182): yii\\queue\\cli\\Queue->handleMessage()","#10 /site/path/site-prod/vendor/craftcms/cms/src/queue/Queue.php(157): craft\\queue\\Queue->executeJob()","#11 [internal function]: craft\\queue\\Queue->craft\\queue\\{closure}()","#12 /site/path/site-prod/vendor/yiisoft/yii2-queue/src/cli/Queue.php(117): call_user_func()","#13 /site/path/site-prod/vendor/craftcms/cms/src/queue/Queue.php(165): yii\\queue\\cli\\Queue->runWorker()","#14 /site/path/site-prod/vendor/craftcms/cms/src/controllers/QueueController.php(82): craft\\queue\\Queue->run()","#15 [internal function]: craft\\controllers\\QueueController->actionRun()","#16 /site/path/site-prod/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()","#17 /site/path/site-prod/vendor/yiisoft/yii2/base/Controller.php(178): yii\\base\\InlineAction->runWithParams()","#18 /site/path/site-prod/vendor/yiisoft/yii2/base/Module.php(552): yii\\base\\Controller->runAction()","#19 /site/path/site-prod/vendor/craftcms/cms/src/web/Application.php(301): yii\\base\\Module->runAction()","#20 /site/path/site-prod/vendor/craftcms/cms/src/web/Application.php(625): craft\\web\\Application->runAction()","#21 /site/path/site-prod/vendor/craftcms/cms/src/web/Application.php(280): craft\\web\\Application->_processActionRequest()","#22 /site/path/site-prod/vendor/yiisoft/yii2/base/Application.php(384): craft\\web\\Application->handleRequest()","#23 /site/path/site-prod/public/index.php(26): yii\\base\\Application->run()","#24 {main}"],"memory":10764544,"exception":"[object] (craft\\db\\QueryAbortedException(code: 0):  at /site/path/site-prod/vendor/craftcms/cms/src/elements/db/ElementQuery.php:1211)"} 
2022-06-02 14:23:42 [queue.ERROR] [craft\db\QueryAbortedException]  {"trace":["#0 /site/path/site-prod/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\\elements\\db\\ElementQuery->prepare()","#1 /site/path/site-prod/vendor/yiisoft/yii2/db/Query.php(157): yii\\db\\QueryBuilder->build()","#2 /site/path/site-prod/vendor/yiisoft/yii2/db/BatchQueryResult.php(159): yii\\db\\Query->createCommand()","#3 /site/path/site-prod/vendor/yiisoft/yii2/db/BatchQueryResult.php(132): yii\\db\\BatchQueryResult->fetchData()","#4 /site/path/site-prod/vendor/yiisoft/yii2/db/BatchQueryResult.php(121): yii\\db\\BatchQueryResult->next()","#5 /site/path/site-prod/vendor/solspace/craft-freeform/packages/plugin/src/Services/SubmissionsService.php(453): yii\\db\\BatchQueryResult->rewind()","#6 /site/path/site-prod/vendor/solspace/craft-freeform/packages/plugin/src/Jobs/PurgeSpamJob.php(14): Solspace\\Freeform\\Services\\SubmissionsService->purgeSubmissions()","#7 /site/path/site-prod/vendor/yiisoft/yii2-queue/src/Queue.php(246): Solspace\\Freeform\\Jobs\\PurgeSpamJob->execute()","#8 /site/path/site-prod/vendor/yiisoft/yii2-queue/src/cli/Queue.php(147): yii\\queue\\Queue->handleMessage()","#9 /site/path/site-prod/vendor/craftcms/cms/src/queue/Queue.php(182): yii\\queue\\cli\\Queue->handleMessage()","#10 /site/path/site-prod/vendor/craftcms/cms/src/queue/Queue.php(157): craft\\queue\\Queue->executeJob()","#11 [internal function]: craft\\queue\\Queue->craft\\queue\\{closure}()","#12 /site/path/site-prod/vendor/yiisoft/yii2-queue/src/cli/Queue.php(117): call_user_func()","#13 /site/path/site-prod/vendor/craftcms/cms/src/queue/Queue.php(165): yii\\queue\\cli\\Queue->runWorker()","#14 /site/path/site-prod/vendor/craftcms/cms/src/controllers/QueueController.php(82): craft\\queue\\Queue->run()","#15 [internal function]: craft\\controllers\\QueueController->actionRun()","#16 /site/path/site-prod/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()","#17 /site/path/site-prod/vendor/yiisoft/yii2/base/Controller.php(178): yii\\base\\InlineAction->runWithParams()","#18 /site/path/site-prod/vendor/yiisoft/yii2/base/Module.php(552): yii\\base\\Controller->runAction()","#19 /site/path/site-prod/vendor/craftcms/cms/src/web/Application.php(301): yii\\base\\Module->runAction()","#20 /site/path/site-prod/vendor/craftcms/cms/src/web/Application.php(625): craft\\web\\Application->runAction()","#21 /site/path/site-prod/vendor/craftcms/cms/src/web/Application.php(280): craft\\web\\Application->_processActionRequest()","#22 /site/path/site-prod/vendor/yiisoft/yii2/base/Application.php(384): craft\\web\\Application->handleRequest()","#23 /site/path/site-prod/public/index.php(26): yii\\base\\Application->run()","#24 {main}"],"memory":10766280,"exception":"[object] (craft\\db\\QueryAbortedException(code: 0):  at /site/path/site-prod/vendor/craftcms/cms/src/elements/db/ElementQuery.php:1211)"} 
```

- Craft Version: 4.0.3
- Freeform Version: 4.0.0-beta.8
- Freeform Edition: PRO
- Fresh Install or Upgrade: Upgrade

**Comment by benjaminkohl** [comment]
The number of spam submissions is actually zero and the spam days setting is currently set to 30 days. Maybe there is an issue with the query when there is no spam.

**Comment by kjmartens** [comment]
This should be fixed now in 4.0.0-beta.11. 🙂
